### ISSUE TITLE
Unable to patch ubuntu kernel 2.6.31-rt

### ISSUE THREAD

**Issue by bogdanroman** [entrope/linux-magicmouse#3]
I've tried applying the new patches (commit 3306fac) to the stock linux-rt-2.6.31 ubuntu kernel, but the first two patches fail:

$ patch --dry-run -p1 < /tmp/MagicMouse/v0.2/0001-Bluetooth-Keep-a-copy-of-each-HID-device-s-report-de.patch 
patching file net/bluetooth/hidp/core.c
Hunk #1 succeeded at 681 (offset -22 lines).
Hunk #2 FAILED at 728.
Hunk #3 FAILED at 776.
Hunk #4 FAILED at 874.
3 out of 4 hunks FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file net/bluetooth/hidp/hidp.h
Hunk #1 succeeded at 152 (offset -2 lines).

$ patch --dry-run -p1 < /tmp/MagicMouse/v0.2/0002-HID-make-raw-reports-possible-for-both-feature-and-o.patch 
patching file drivers/hid/hidraw.c
patching file drivers/hid/usbhid/hid-core.c
Hunk #1 succeeded at 775 (offset 1 line).
Hunk #2 succeeded at 787 (offset 1 line).
patching file include/linux/hid.h
Hunk #1 FAILED at 501.
1 out of 1 hunk FAILED -- saving rejects to file include/linux/hid.h.rej
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 313.
1 out of 1 hunk FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej

The third and forth seem to work just fine.
The patches prior to the last commit (3306fac) worked fine on this version of the kernel.

**Comment by bogdanroman** [comment]
I'm guessing the current patches can be applied to the mainstream version of the kernel. What would one have to do to get the latest modifications on a older kernel snapshot ? i.e: 2.6.31 ?

**Comment by entrope** [comment]
The patch series that I uploaded on the 11th (commit cd7fe9f) should include everything you need for 2.6.31- and 2.6.32-based kernels.  Does that set work for you?

**Comment by bogdanroman** [comment]
This patch series still has hunks that fail, and since right now I'm not willing to change the version of the kernel I'm using, I will try to create a series of patches based on the modifications made from 2.6.31 to 2.6.32. I will post a comment when/if I succeed.

I realize this is not an actual issue, since supporting patches for several kernel versions can be very time consuming, so probably it would be best to close this one.

**Comment by entrope** [comment]
Since there's been no action lately, I'm going to close this.

**Comment by alexleach** [comment]
Hello. I'm running the latest 64bit version of Ubuntu (Linux 2.6.32-24-generic #41-Ubuntu SMP x86_64 GNU/Linux) and it looks like all the hunks fail to patch, having followed the install instructions... I don't even know what a hunk is, so could do with some help on this. I thought sudo privileges might help, but no...
Output:-
root@metabuntu:/home/albl500/Downloads/linux-magicmouse# sh
# pwd
/home/albl500/Downloads/linux-magicmouse
# for PATCH in *.patch ; do patch -p1 < $PATCH ; done
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 280.
Hunk #2 FAILED at 785.
2 out of 2 hunks FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 243.
Hunk #2 FAILED at 282.
Hunk #3 FAILED at 307.
3 out of 3 hunks FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file drivers/hid/hidraw.c
Hunk #1 FAILED at 134.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/hidraw.c.rej
patching file drivers/hid/usbhid/hid-core.c
Hunk #1 FAILED at 776.
Hunk #2 FAILED at 787.
2 out of 2 hunks FAILED -- saving rejects to file drivers/hid/usbhid/hid-core.c.rej
patching file include/linux/hid.h
Hunk #1 FAILED at 501.
1 out of 1 hunk FAILED -- saving rejects to file include/linux/hid.h.rej
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 313.
1 out of 1 hunk FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 783.
1 out of 1 hunk FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file net/bluetooth/hidp/core.c
Hunk #1 FAILED at 714.
Hunk #2 FAILED at 781.
Hunk #3 FAILED at 817.
Hunk #4 FAILED at 911.
4 out of 4 hunks FAILED -- saving rejects to file net/bluetooth/hidp/core.c.rej
patching file net/bluetooth/hidp/hidp.h
Hunk #1 FAILED at 154.
1 out of 1 hunk FAILED -- saving rejects to file net/bluetooth/hidp/hidp.h.rej
patching file drivers/hid/hid-core.c
Hunk #1 FAILED at 53.
Hunk #2 FAILED at 77.
2 out of 2 hunks FAILED -- saving rejects to file drivers/hid/hid-core.c.rej
patching file include/linux/hid.h
Hunk #1 FAILED at 690.
1 out of 1 hunk FAILED -- saving rejects to file include/linux/hid.h.rej
patching file drivers/hid/Kconfig
Hunk #1 FAILED at 183.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/Kconfig.rej
patching file drivers/hid/Makefile
Hunk #1 FAILED at 31.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/Makefile.rej
patching file drivers/hid/hid-core.c
Hunk #1 FAILED at 1256.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/hid-core.c.rej
patching file drivers/hid/hid-ids.h
Hunk #1 FAILED at 56.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/hid-ids.h.rej
The next patch would create the file drivers/hid/hid-magicmouse.c,
which already exists!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
1 out of 1 hunk ignored
patching file drivers/hid/Kconfig
Hunk #1 FAILED at 184.
1 out of 1 hunk FAILED -- saving rejects to file drivers/hid/Kconfig.rej
patching file drivers/hid/hid-magicmouse.c
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
8 out of 8 hunks ignored -- saving rejects to file drivers/hid/hid-magicmouse.c.rej

**Comment by alexleach** [comment]
I figured out to use patch it within the directory /usr/src/linux-headers-2.6.32-24-generic, and so from there, I ran the command:-
for PATCH in /home/albl500/Downloads/linux-magicmouse/*.patch ; do patch -p1 < $PATCH ; done

Getting similar output to above.. I've compiled and installed loads of stuff before, but I've never messed with the kernel. I've also rebooted (after adding "hid-magicmouse" to the /etc/modules file) after running the above command from the linux-headers directory. Some seemed to succeed whilst others failed.
Succeeded:
drivers/hid/Kconfig
include/linux/hid.h
drivers/hid/Makefile
The rest failed. Will reboot and see what happens...
